### PR TITLE
fix macos, test on the latest clang

### PR DESCRIPTION
**Pull request by aminya**

### CODE REVIEW

File: .github/workflows/d.yml [hunk 1]
```diff
@@ -42,18 +50,19 @@ jobs:
...
           version: ${{ matrix.clang }}
+          cached: ${{ steps.cache.outputs.cache-hit }}
 
       - name: Linux - link libclang.so
         if: contains(matrix.os, 'ubuntu')
         run: sudo ln -s libclang-${{ matrix.clang }}.so.1 /lib/x86_64-linux-gnu/libclang.so
         working-directory: ${{ env.LLVM_PATH }}/lib
 
       - name: Run tests
+        continue-on-error: ${{ contains(matrix.os, 'macos') }}
```
Comment (aminya) [inline .github/workflows/d.yml:65]: 28 tests fail on macOS

Comment (aminya) [reply .github/workflows/d.yml:65]: This commit reduces the number of failing tests to 9 on macOS 😄 
https://github.com/atilaneves/dpp/pull/292/commits/284f294f70d76afc5611b1cfcaa49e6807276fdc

**Review by aminya — COMMENTED** [review]
Now, the macos failures are related to the linker:
```d
it.c.run.c.function named debug:
    tests/it/package.d:288 - Could not execute `clang -fPIE -o c.o -g -std=c11 -c c.c`:
c.c:2:26: fatal error: 'stdio.h' file not found
                #include <stdio.h>
                         ^~~~~~~~~
1 error generated.
```

**Comment by aminya** [comment]
By following these instructions I can finally use dpp on my Mac VM:
https://github.com/atilaneves/dpp/pull/292/commits/53151891328fb6b95de207728a545cfcb6be0ae8

File: .github/workflows/d.yml (hunk 1 above)
Comment (atilaneves) [reply .github/workflows/d.yml:81]: I'd say it'd be better to fix all of them first - otherwise this is sweeping dust under the rug.

Comment (aminya) [reply .github/workflows/d.yml:81]: This PR fixes the issues with generating the headers, which is all I need. I don't use dpp as my compiler. I only use it in `preprocess-headers` mode.

Comment (atilaneves) [reply .github/workflows/d.yml:81]: But it wouldn't be a good idea for things to work differently on MacOS.

**Review by atilaneves — APPROVED** [review]

**Pull request merged**